tests/main/uc20-create-partitions/task.yaml

```diff
@@ -138,29 +138,33 @@ execute: |
     rm ./mnt/EFI/boot/grubx64.efi
     umount ./mnt
 
-    echo "now add a partition"
+    echo "Add a non-creatable partition to see it fail installation"
+    cp gadget-dir/meta/gadget.yaml gadget.yaml.backup
     cat >> gadget-dir/meta/gadget.yaml <<EOF
           - name: Other ext4
             type: 83,0FC63DAF-8483-4772-8E79-3D69D8477DE4
             filesystem: ext4
             filesystem-label: other-ext4
             size: 110M
     EOF
+    uc20-create-partitions ./gadget-dir "$kerneldir" "$LOOP" 2>&1 | MATCH "panic: gadget and ${LOOP} partition table not compatible: cannot find gadget structure #6 \(\"Other ext4\"\) on disk"
+
+    # replace the gadget.yaml without the extra partition
+    cp gadget.yaml.backup gadget-dir/meta/gadget.yaml
+
+    echo "Ensure we can deploy with mounting"
     uc20-create-partitions --mount ./gadget-dir "$kerneldir" "$LOOP"
     sfdisk -l "$LOOP" | MATCH "${LOOP}p1 .* 1M\s* BIOS boot"
     sfdisk -l "$LOOP" | MATCH "${LOOP}p2 .* 1.2G\s* EFI System"
     sfdisk -l "$LOOP" | MATCH "${LOOP}p3 .* 750M\s* Linux filesystem"
     sfdisk -l "$LOOP" | MATCH "${LOOP}p4 .* 16M\s* Linux filesystem"
     sfdisk -l "$LOOP" | MATCH "${LOOP}p5 .* 1G\s* Linux filesystem"
```

Comment:
This test now fails with:
```
2021-11-25T11:36:47.1924969Z ##[error]2021-11-25 11:36:47 Error executing google:ubuntu-20.04-64:tests/main/uc20-create-partitions (nov251058-167771) : 
...
2021-11-25T11:36:47.2060148Z + MATCH '/dev/loop1p5 .* 1G\s* Linux filesystem'
2021-11-25T11:36:47.2060872Z + sfdisk -l /dev/loop1
2021-11-25T11:36:47.2061387Z grep error: pattern not found, got:
2021-11-25T11:36:47.2061990Z Disk /dev/loop1: 18.64 GiB, 20000000000 bytes, 39062500 sectors
2021-11-25T11:36:47.2062538Z Units: sectors of 1 * 512 = 512 bytes
2021-11-25T11:36:47.2063128Z Sector size (logical/physical): 512 bytes / 512 bytes
2021-11-25T11:36:47.2063804Z I/O size (minimum/optimal): 512 bytes / 512 bytes
2021-11-25T11:36:47.2064354Z Disklabel type: gpt
2021-11-25T11:36:47.2065278Z Disk identifier: 9F0C4508-FC05-AD44-890E-AD7CC68ABA34
2021-11-25T11:36:47.2065783Z 
2021-11-25T11:36:47.2066249Z Device         Start      End  Sectors  Size Type
2021-11-25T11:36:47.2067148Z /dev/loop1p1    2048     4095     2048    1M BIOS boot
2021-11-25T11:36:47.2067681Z /dev/loop1p2    4096  2461695  2457600  1.2G EFI System
2021-11-25T11:36:47.2068267Z /dev/loop1p3 2461696  3997695  1536000  750M Linux filesystem
2021-11-25T11:36:47.2068887Z /dev/loop1p4 3997696  4030463    32768   16M Linux filesystem
2021-11-25T11:36:47.2077664Z /dev/loop1p5 4030464 39062466 35032003 16.7G Linux filesystem
...
```

Reply:
ah indeed the MATCH needs to be updated since now the last partition is expanded

Reply:
uc20-create-partitions is happy now